go-controller/pkg/libovsdbops/router.go

```diff
@@ -451,15 +365,33 @@ func AddOrUpdateNATsToRouter(nbClient libovsdbclient.Client, routerName string,
 	return err
 }
 
-func DeleteNATsFromRouter(nbClient libovsdbclient.Client, routerName string, nats ...*nbdb.NAT) error {
-	router := &nbdb.LogicalRouter{Name: routerName}
-	ops, err := DeleteNATsFromRouterOps(nbClient, nil, router, nats...)
-	if err != nil {
-		return err
+// DeleteNATsWithPredicateOps looks up NATs from the cache based on a given
+// predicate, deletes them, removes them from the provided logical router and
+// returns the corrsponding ops
+func DeleteNATsWithPredicateOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, routerName string, p natPredicate) ([]libovsdb.Operation, error) {
+	router := &nbdb.LogicalRouter{
+		Name: routerName,
 	}
 
-	_, err = TransactAndCheck(nbClient, ops)
-	return err
+	deleted := []*nbdb.NAT{}
+	opModels := []OperationModel{
+		{
+			ModelPredicate: p,
+			ExistingResult: &deleted,
+			DoAfter:        func() { router.Nat = ExtractUUIDsFromModels(&deleted) },
+			BulkOp:         true,
+		},
+		{
+			Model:            router,
+			ModelPredicate:   func(lr *nbdb.LogicalRouter) bool { return lr.Name == routerName },
+			OnModelMutations: []interface{}{&router.Nat},
+			ErrNotFound:      true,
+			BulkOp:           false,
+		},
+	}
+
+	m := NewModelClient(nbClient)
+	return m.DeleteOps(ops, opModels...)
 }
 
 // GATEWAY CHASSIS OPs
```

Comment:
Super NIT alert: since you are modifying this file, could you please fix the spelling of `associated` on line 414?

Reply:
Sure, will include that in the relevant commit.